File: dataset/codenet/Python/Code/s952876397.py
n = int(input())
s = input()

left = 0
for i in s:
    if i == ")":
        left += 1
    else:
        break

left1 = 0
right1 = 0

for i in range(left, n):
    if s[i] == "(":
        right1 += 1
    elif right1 > 0:
        right1 -= 1
    else:
        left1 += 1

left1 += left
for i in range(right1):
    s += ")"
for i in range(left1):
    s = "(" + s

print(s)

File: output/qwen_2.5_72b_instruct/codenet/algo_based_translation/python/java/s952876397.java

import java.util.Scanner;

public class s952876397 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        String s = scanner.next();
        
        int left = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == ')') {
                left++;
            } else {
                break;
            }
        }
        
        int left1 = 0;
        int right1 = 0;
        
        for (int i = left; i < n; i++) {
            char c = s.charAt(i);
            if (c == '(') {
                right1++;
            } else if (right1 > 0) {
                right1--;
            } else {
                left1++;
            }
        }
        
        left1 += left;
        for (int i = 0; i < right1; i++) {
            s += ")";
        }
        for (int i = 0; i < left1; i++) {
            s = "(" + s;
        }
        
        System.out.println(s);
    }
}

